### PR TITLE
Add sanitize function in obs-api

### PR DESCRIPTION
**Pull request by HuseinA**
- [x] sanitize char in admin OBS-API
- [x] sanitize char in bucket OBS-API
- [x] sanitize char in object OBS-API
- [x] add naming guidelines in documentation

### CODE REVIEW

File: obs/api/app/controllers/api/storage.py [hunk 1]
```diff
@@ -266,6 +272,9 @@ def post(self, bucket_name):
         object_name = args["object_name"] if args["object_name"] else filename
 
         try:
+            regex = r"[\"\{}^%`\]\[~<>|#]|[\x80-\xFf]"
```
Comment (efenfauzi) [inline obs/api/app/controllers/api/storage.py:275]: regex replace matching char for **Object Name Rules** 

set [**\"\{}^%`\]\[~<>|#**]

match special char **"** (quotation)
match special char **{** (curly brace open)
match **}** (curly brace end)
match **^** (caret)
match **%** (percentage) 
match **`** (grave acent)
match special char **]** (square bracket end)
match special char **[** (square bracket opening)
match **~** (tilde)
match **<** (less than)
match **>** (greater than)
match **|** (vertical bar)
match **#** (pound)
**OR** 
set [**\x80-\xFf**]
character in the range between (index 128) and (index 255) in ASCII Code

File: obs/api/app/controllers/api/storage.py [hunk 2]
```diff
@@ -104,6 +105,11 @@ def post(self, bucket_name):
         secret_key = args["secret_key"].replace(" ", "+")
 
         try:
+            regex = r"[^a-z0-9.-]"
```
[the remark this answers is not in the document]
Comment (HuseinA) [reply obs/api/app/controllers/api/storage.py:108]: some of error has been handled from third party (boto3)

File: obs/api/app/controllers/api/admin.py [hunk 3]
```diff
@@ -12,6 +14,21 @@ def get_client():
...
 
 
+def sanitize(args):
+    arg = {}
+    regex = r"[<>`;|&#]|[\\n]{2}|[%26]{3}|\n"
+    regexuid = r"[^a-z0-9.-]"
+
+    for key, value in args.items():
+        if key == "userId":
+            arg[key] = re.sub(regexuid, "", value)
```
[the remark this answers is not in the document]
Comment (HuseinA) [reply obs/api/app/controllers/api/admin.py:24]: regex has been changed for user id

File: obs/api/app/controllers/api/storage.py [hunk 4]
```diff
@@ -107,9 +107,13 @@ def post(self, bucket_name):
         try:
             regex = r"[^a-z0-9.-]"
             bucket_name = re.sub(regex, "", bucket_name)
-            if len(bucket_name) < 3:
-                return response(400, f"'{bucket_name}' too short for bucket name")
+            current_app.logger.error(len(bucket_name))
+            if not 2 < len(bucket_name) < 64:
```
Comment (efenfauzi) [inline obs/api/app/controllers/api/storage.py:111]: pass check bucket name between 3 and 63 char. 
* Bucket names must be at least 3 and no more than 63 characters long.

File: obs/api/app/controllers/api/admin.py [hunk 5]
```diff
@@ -17,7 +17,7 @@ def get_client():
 def sanitize(args):
     arg = {}
     regex = r"[<>`;|&#]|[\\n]{2}|[%26]{3}|\n"
-    regexuid = r"[^a-z0-9.-]"
+    regexuid = r"[^\w\d@#_\-.]"
```
Comment (efenfauzi) [inline obs/api/app/controllers/api/admin.py:20]: \w anyword match a-zA-Z0-9_
\d match desimal number 0-9
match char **@** 
match char **#**
match char **_** 
match spesifik char **-**
match char **.**

File: obs/api/app/controllers/api/storage.py [hunk 6]
```diff
@@ -272,7 +276,7 @@ def post(self, bucket_name):
         object_name = args["object_name"] if args["object_name"] else filename
 
         try:
-            regex = r"[\"\{}^%`\]\[~<>|#]|[\x80-\xFf]"
+            regex = r"[\"\{}^%`\]\[~<>|#]|[^\x00-\x7F]"
```
Comment (efenfauzi) [inline obs/api/app/controllers/api/storage.py:279]: ^\x00-\x7F
match any char between 0 and 127

**Pull request merged**